CodeConverter/CSharp/ExpressionNodeVisitor.cs

```diff
@@ -295,7 +295,7 @@ public override async Task<CSharpSyntaxNode> VisitFinallyBlock(VBasic.Syntax.Fin
     public override async Task<CSharpSyntaxNode> VisitCTypeExpression(VBasic.Syntax.CTypeExpressionSyntax node)
     {
         var csharpArg = await node.Expression.AcceptAsync<ExpressionSyntax>(TriviaConvertingExpressionVisitor);
-        var typeInfo = _semanticModel.GetTypeInfo(node);
+        var typeInfo = _semanticModel.GetTypeInfo(node.Type);
```

Comment:
The original here was likely me trying to skip unnecessary casts, but as shown below, that's not always correct, and as you say, there are other ways to simplify code that people should probably use after conversion.